src/core/exec-invoke.c

```diff
@@ -3162,6 +3167,9 @@ static int apply_mount_namespace(
                 if (asprintf(&extension_dir, "/run/user/" UID_FMT "/systemd/unit-extensions", geteuid()) < 0)
                         return -ENOMEM;
 
+                if (asprintf(&private_tmp_dir, "/run/user/" UID_FMT "/systemd/unit-private-tmp", geteuid()) < 0)
+                        return -ENOMEM;
```

Comment:
hmm, can we maybe merge these two fields into one? i.e. just pass around a single field `private_namespace_dir` or so, which carries the common prefix of these two paths, i.e. without the `/unit-extension` and `/unit-private-tmp` suffix.

And then, append that suffix when you actually use it.

The structure already has too many fields, and I sense that sooner or later we will need more dirs like that, and then it would be even more pressing to avoid an inflation of even more fields like this.

Reply:
ok, done